layer.md

```diff
@@ -1,63 +1,129 @@
-# Creating an Image Filesystem Changeset
+# Filesystem Layer Changeset
 
-An example of creating an Image Filesystem Changeset follows.
+A layer changeset is the distributable difference between filesystems.
+A layer changeset is not directly extracted, but applied to a filesystem.
 
-An image root filesystem is first created as an empty directory.
-Here is the initial empty directory structure for a changeset using the randomly-generated directory name `c3167915dc9d` ([actual layer DiffIDs are generated based on the content](#id_desc)).
+## Distributable Format
+
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST be packaged in [tar archive][tar-archive].
+Layer Changesets for the [mediatype](./media-types.md) `application/vnd.oci.image.layer.tar+gzip` MUST NOT include duplicate entries for file paths in the resulting [tar archive][tar-archive].
+
+## Change Types
+
+Types of changes that can occur in a changeset are:
+
+* Additions
+* Modifications
+* Removals
+
+Additions and Modifications are reflected the same in the changeset tar archive.
+
+Removals are reflected using "[whiteout](#whiteouts)" file entries (See [Representing Changes](#representing-changes)).
+
+### File Attributes
+
+File attributes that are provided for Additions and Modifications include:
+
+* Modification Time (`mtime`)
+* User ID (`uid`)
+ * User Name (`uname`) is secondary to `uid`
+* Group ID (`gid `)
+ * Group Name (`gname`) is secondary to `gid`
+* Mode (`mode`)
+* Extended Attributes (`xattrs`)
+
+[Sparse files](https://en.wikipedia.org/wiki/Sparse_file) SHOULD NOT be used.
+Due to lack of consistent support will appear to mutate regular files on creating or applying of a layer changeset.
+
+## Creating
+
+### Initial Root Filesystem
+
+An image root filesystem has an initial state as an empty directory.
+The name of the directory is not relevant to the layer itself, only for the purpose of producing comparisons.
+
+Here is an initial empty directory structure for a changeset, with a unique directory name `rootfs-c9d-v1`.
 
 ```
-c3167915dc9d/
+rootfs-c9d-v1/
```

Comment:
On Wed, Sep 07, 2016 at 06:55:36AM -0700, Vincent Batts wrote:
> >  
> >  ```
> > -c3167915dc9d/
> > +rootfs-c9d-v1/
> 
> uniqueness. We could call it `fart-jar/`. 

I don't have and of rootfs, rootfs-v1, rootfs-c9d-v1, or fart-jar in
my home directory ;).  I think rootfs (where you have rootfs-c9d-v1)
and rootfs-v1 (where you have rootfs-c9d-v1.s1) are best because:

a. They include ‘rootfs’, which is where layers will be unpacked by
   default [1].
b. They distinguish between the mutable working directory (rootfs) and
   a snapshot of it (rootfs-v1) to make it easy to talk about changes.

[1]: https://github.com/opencontainers/image-spec/blame/d3cfb991ee3ff9e7c9570027cb9696213cfcee00/cmd/oci-image-tool/man/oci-image-tool-create-runtime-bundle.1.md#L24